File: golang/Sorting Algorithms/Bubble Sort/bubbleSort.go
package main

import "fmt"

func main() {
	arr := []int{0, 5, 3, 2, 8, 4, 16, 29,1 4, 4, 7}
	fmt.Printf("Array before Bubble Sort\t->\t %v\n", arr)

	bubbleSort(arr)

	fmt.Printf("Array after Bubble Sort\t->\t %v\n", arr)
}

func bubbleSort(arr []int) []int {
	for i := 0; i < len(arr)-1; i++ {
		for j := 0; j < len(arr)-i-1; j++ {
			if arr[j] > arr[j+1] {
				var temp = arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
			}
		}
	}
	return arr
}

File: Dart/Sorting Algorithms/Bubble Sort/bubbleSort.dart
main() {
  List<int> arr = [10, 2, 3, 21, 12, 8, 32, 22, 13, 6];
  print("Array before bubble sort \t->\t$arr");
  print("Array after bubble sort \t->\t${bubbleSort(arr)}");
}

List<int> bubbleSort(List<int> arr) {
  for (var i = 0; i < arr.length - 1; i++) {
    for (var j = 0; j < arr.length - 1 - i; j++) {
      if (arr[j] > arr[j + 1]) {
        var temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
  }
  return arr;
}
